main.cpp

```diff
@@ -1,23 +1,134 @@
 #include <bits/stdc++.h>
+
+#define nol nums_of_lfs
+
 using namespace std;
 
-int n;
-vector <char> cars;
+struct lf
+{
+    int indx, val;
+};
 
-int main()
+bool operator<(lf l, lf r)
+    { return (l.val != r.val ? l.val < r.val : l.indx < r.indx); }
```

Comment:
use pair instead, with default comparator. Firstly operators overloading is an anti-patern usually, secondly it doesn't works here. To solve such problems use google: https://stackoverflow.com/questions/2620862/using-custom-stdset-comparator